observer/retry.go

```diff
@@ -0,0 +1,31 @@
+package observer
+
+import (
+	"github.com/trustwallet/blockatlas"
+	"math/rand"
+	"time"
+)
+
+type GetBlockByNumber func(num int64) (*blockatlas.Block, error)
+
+type stop struct {
+	error
+}
+
+func retry(attempts int, sleep time.Duration, f GetBlockByNumber, n int64) (*blockatlas.Block, error) {
+	r, err := f(n)
+	if err != nil {
+		if s, ok := err.(stop); ok {
+			return nil, s.error
+		}
+		if attempts--; attempts > 0 {
+			// Add some randomness to prevent creating a Thundering Herd
+			jitter := time.Duration(rand.Int63n(int64(sleep)))
+			sleep = sleep + jitter/2
+
+			time.Sleep(sleep)
+			return retry(attempts, sleep*2, f, n)
```

Comment:
Code formatting.

Reply:
This file is using the default format for Golang, `go fmt`. They automatically arrange the var with `*2` without spaces. This pattern is used by default in Golang.  If we prefer to follow this rule, I can change this for Goland stop to do this automatically. What do you think?